compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -962,6 +962,44 @@ static LogicalResult setFftConfig(mlir::FunctionOpInterface entryPoint,
       workgroupSize);
 }
 
+//===----------------------------------------------------------------------===//
+// Winograd Pipeline Configuration
+//===----------------------------------------------------------------------===//
+template <typename WinogradOp>
+static LogicalResult setWinogradOpConfig(mlir::FunctionOpInterface entryPoint,
+                                         WinogradOp op,
+                                         const TargetInfo &targetInfo) {
+  static_assert(
+      std::is_same<WinogradOp, IREE::LinalgExt::WinogradInputTransformOp>() ||
+          std::is_same<WinogradOp,
+                       IREE::LinalgExt::WinogradFilterTransformOp>() ||
+          std::is_same<WinogradOp,
+                       IREE::LinalgExt::WinogradOutputTransformOp>(),
```

Comment:
there should be `std::is_same_v`, no? I'm surprised this compiles

Reply:
you can also simplify this with `llvm::is_one_of`

Reply:
OK I know why, `std::integral_constant` has a constexpr conversion operator for the value type: https://en.cppreference.com/w/cpp/types/integral_constant

Reply:
OK I see why, you are constructing integral constants and then the conversion operator for the value type kicks in: https://en.cppreference.com/w/cpp/types/integral_constant